plugins/modules/azure_rm_virtualnetworkgatewayconnection.py

```diff
@@ -0,0 +1,304 @@
+  express_route_gateway_bypass:
+    description: Bypass ExpressRoute Gateway for data forwarding.
+    type: bool
+extends_documentation_fragment:
+    - azure.azcollection.azure
+    - azure.azcollection.azure_tags
+'''
+
+EXAMPLES = '''
+    - name: Create Virtual Network Gateway Connection
+      azure.azcollection.azure_rm_virtualnetworkgatewayconnection:
+        name: Test-VN-GW-Conn
+        location: West US 3
+        resource_group: VPN-RG
+        virtual_network_gateway1: VPN-VNG
+        local_network_gateway: Local-NW-GW
+        connection_type: IPsec
+        connection_protocol: IKEv2
+        shared_key: SuperSecretKey
+        state: present
+
+    - name: Delete Virtual Network Gateway Connections
+      azure.azcollection.azure_rm_virtualnetworkgatewayconnection:
+        name: Test-VN-GW-Conn
+        location: West US 3
+        resource_group: VPN-RG
+        state: absent
+'''
+
+RETURN = '''
+id:
+    description:
+        - Virual Network Gateway connection resource ID.
+    returned: always
+    type: str
+    sample: "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.Network/connections/MyVNGWConnection"
+'''
+try:
+    from azure.core.exceptions import ResourceNotFoundError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+
+
+def vngwconn_to_dict(vngwconn):
+    results = dict(
+        id=vngwconn.id,
+        name=vngwconn.name,
+        location=vngwconn.location,
+        etag=vngwconn.etag,
+        tags=vngwconn.tags,
+        virtual_network_gateway1=vngwconn.virtual_network_gateway1,
+        virtual_network_gateway2=vngwconn.virtual_network_gateway2,
+        local_network_gateway=vngwconn.local_network_gateway2,
+        authorization_key=vngwconn.authorization_key,
+        connection_type=vngwconn.connection_type,
+        connection_protocol=vngwconn.connection_protocol,
+        routing_weight=vngwconn.routing_weight,
+        dpd_timeout_seconds=vngwconn.dpd_timeout_seconds,
+        shared_key=vngwconn.shared_key,
+        enable_bgp=vngwconn.enable_bgp,
+        use_local_azure_ip_address=vngwconn.use_local_azure_ip_address,
+        use_policy_based_traffic_selectors=vngwconn.use_policy_based_traffic_selectors,
+        express_route_gateway_bypass=vngwconn.express_route_gateway_bypass,
+        provisioning_state=vngwconn.provisioning_state,
+    )
+    return results
+
+
+class AzureRMVirutalNetworkGatewayConnection(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            name=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            location=dict(type='str'),
+            virtual_network_gateway1=dict(type='str', required=False),
```

Comment:
As far I can see, virtual_network_gateway is required for "IPsec", "Vnet2Vnet" and "ExpressRoute"; not sure about  "VPNClient" which the azure console does not offer as an option.